app/assets/javascripts/controllers/security_group/security_group_form_controller.js

```diff
@@ -1,54 +1,99 @@
 ManageIQ.angular.app.controller('securityGroupFormController', ['$scope', 'securityGroupFormId', 'miqService', 'API', function($scope, securityGroupFormId, miqService, API) {
-  $scope.securityGroupModel = { name: '' };
-  $scope.formId = securityGroupFormId;
-  $scope.afterGet = false;
-  $scope.modelCopy = angular.copy( $scope.securityGroupModel );
-  $scope.model = "securityGroupModel";
-
-  ManageIQ.angular.scope = $scope;
-
-  if (securityGroupFormId == 'new') {
-    $scope.securityGroupModel.name = "";
-    $scope.securityGroupModel.description = "";
-    $scope.newRecord = true;
-  } else {
-    miqService.sparkleOn();
-    API.get("/api/security_groups/" + securityGroupFormId + "?attributes=cloud_tenant").then(function(data) {
-      $scope.afterGet = true;
-      $scope.securityGroupModel.name = data.name;
-      $scope.securityGroupModel.description = data.description;
-      $scope.securityGroupModel.cloud_tenant_name = data.cloud_tenant.name;
-      $scope.modelCopy = angular.copy( $scope.securityGroupModel );
+  var vm = this;
+
+  var init = function() {
+    vm.securityGroupModel = {
+      name: "",
+      description: "",
+      cloud_tenant_name: "",
+      firewall_rules: [],
+    };
+
+    getSecurityGroups();
+    vm.hostProtocols = ["", "TCP", "UDP", "ICMP"];
+    vm.networkProtocols = ["IPV4", "IPV6"];
+    vm.directions = ["inbound", "outbound"];
+
+    vm.afterGet = false;
+    vm.formId = securityGroupFormId;
+    vm.model = "securityGroupModel";
+    vm.newRecord = securityGroupFormId == "new";
+    vm.saveable = miqService.saveable;
+    ManageIQ.angular.scope = $scope;
+
+    if (vm.newRecord) {
+      vm.securityGroupModel.name = "";
+    } else {
+      miqService.sparkleOn();
+      API.get("/api/security_groups/" + securityGroupFormId + "?attributes=cloud_tenant,firewall_rules").then(function(data) {
+        vm.securityGroupModel.name = data.name;
+        vm.securityGroupModel.description = data.description;
+        vm.securityGroupModel.cloud_tenant_name = data.cloud_tenant.name;
+        vm.securityGroupModel.firewall_rules = data.firewall_rules;
+        vm.afterGet = true;
+      }).catch(miqService.handleFailure);
       miqService.sparkleOff();
+    }
+    vm.modelCopy = angular.copy( vm.securityGroupModel );
```

Comment:
Move the `angular.copy` line after `vm.afterGet` since that's when we get data from the API.

(Otherwise `Reset` would empty all values in `vm.securityGroupModel` that the API populated)